src/Featurizers/Components/WindowFeaturizerBase.h

```diff
@@ -344,6 +346,12 @@ typename CircularBuffer<T>::const_iterator CircularBuffer<T>::cend() const {
     return end_impl<const_iterator>();
 }
 
+template <class T>
+void CircularBuffer<T>::clear () {
+    _start_offset = 0;
+    _data.clear();
```

Comment:
>_data.clear(); [](start = 4, length = 14)

When data is cleared from a vector, I believe that its capacity is reset as well. We probably want to reserve space as what is currently being done in the constructor. In fact, the constructor should probably call this method directly and remove any duplicated code.

Reply:
clear() will be added in a separate PR.